include/wlr/types/wlr_surface_layers.h

```diff
@@ -0,0 +1,71 @@
+#ifndef _WLR_SURFACE_LAYERS_H
+#define _WLR_SURFACE_LAYERS_H
+
+#include <wayland-server.h>
+
+struct wlr_surface_layers {
+	struct wl_global *wl_global;
+
+	// list of wlr_layer_surface elements, ordered by ascending layer
+	struct wl_list surfaces;
+
+	struct {
+		struct wl_signal new_surface;
+	} events;
+
+	void *data;
+};
+
+enum wlr_layer_surface_invalid {
+	WLR_LAYER_SURFACE_INVALID_INTERACTIVITY = 1,
+	WLR_LAYER_SURFACE_INVALID_ANCHOR = 2,
+	WLR_LAYER_SURFACE_INVALID_EXCLUSIVE_ZONE = 4,
+	WLR_LAYER_SURFACE_INVALID_MARGIN = 8,
+};
+
+struct wlr_layer_surface_state {
+	uint32_t invalid; // wlr_layer_surface_invalid
+	uint32_t input_types, exclusive_types; // layer_surface_input_device
+	uint32_t anchor; // layer_surface_anchor
+	uint32_t exclusive_zone;
+	int32_t margin_horizontal, margin_vertical;
+};
+
+struct wlr_layer_surface {
+	struct wl_resource *resource;
+	struct wlr_surface_layers *surface_layers;
+	struct wlr_surface *surface;
+	struct wlr_output *output;
+	uint32_t layer; // surface_layers_layer
+	struct wl_list link; // wlr_surface_layers.surfaces
+
+	struct wlr_layer_surface_state *current, *pending;
+
+	struct wl_listener surface_destroy_listener;
+	struct wl_listener surface_commit_listener;
+
+	struct {
+		struct wl_signal destroy;
+		struct wl_signal commit;
+	} events;
+
+	void *data;
```

Comment:
What is this for?

Reply:
User-defined data, as in [other](https://github.com/swaywm/wlroots/blob/master/include/wlr/types/wlr_xdg_shell_v6.h#L18) [parts](https://github.com/swaywm/wlroots/blob/master/include/wlr/types/wlr_xdg_shell_v6.h#L135) [of](https://github.com/swaywm/wlroots/blob/master/include/wlr/types/wlr_output.h#L64) [the](https://github.com/swaywm/wlroots/blob/master/include/wlr/types/wlr_screenshooter.h#L19) [code](https://github.com/swaywm/wlroots/search?utf8=%E2%9C%93&q=void+*data&type=).